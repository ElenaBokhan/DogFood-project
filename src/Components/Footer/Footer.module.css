.footer {
    height: 194px;
    display: flex;
    padding: 40px 0;
    justify-content: space-between;
    align-items: center;
}

.footer ul {
    list-style-type: none;
}

.footer ul li {
    line-height: 32px;
}

.footer ul li:hover {
    font-weight: bold;
}

.logo-block {
    display: flex;
    flex-direction: column;
    height: 100%;
    justify-content: space-between;
}

.contacts {
    display: flex;
    flex-direction: column;
    height: 100%;
    width: 168px;
    justify-content: space-between;
}

.social {
    display: flex;
    justify-content: space-between;
    justify-self: flex-end;
}
