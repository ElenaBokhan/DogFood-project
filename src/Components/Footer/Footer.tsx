import instagramIcon from 'assets/logo-instagram.svg';
import vkIcon from 'assets/logo-vk.svg';
import Logo from 'assets/Logo.svg';
import tgIcon from 'assets/telegram 1.svg';
import viberIcon from 'assets/viber 1.svg';
import whatsappIcon from 'assets/whatsapp 1.svg';
import FooterContainer, {EContainerType} from 'Components/Common/Container/Container';
import {IconButton} from 'Components/Common/IconButton/IconButton';
import {EFontType, ETextSize, Text} from 'Components/Common/Text/Text';
import styles from 'Components/Footer/Footer.module.css';

export const Footer = () => {
    const renderFooterLogo = () => {
        return (
            <div className={styles.logoBlock}>
                <img alt="logo" src={Logo} />
                <Text size={ETextSize.S9} value={'© «Интернет-магазин DogFood.ru»'} />
            </div>
        );
    };
    const renderMenu = () => {
        const menu = [
            ['Каталог', 'Акции', 'Новости', 'Отзывы'],
            ['Оплата и доставка', 'Часто спрашивают', 'Обратная связь', 'Контакты'],
        ];
        return (
            <>
                {menu.map((column, index) => {
                    return (
                        <ul key={index}>
                            {column.map((menuItem) => {
                                return (
                                    <li key={menuItem}>
                                        <Text size={ETextSize.S14} value={menuItem} />
                                    </li>
                                );
                            })}
                        </ul>
                    );
                })}
            </>
        );
    };

    const renderContactsGroup = () => {
        return (
            <div className={styles.contacts}>
                <Text size={ETextSize.S16} type={EFontType.GENERAL} value={'Мы на связи'} />
                <Text size={ETextSize.S16} value={'8 (999) 00-00-00'} />
                <Text size={ETextSize.S12} value={'dogfood.ru@gmail.com'} />
                <div className={styles.social}>
                    <IconButton alt="telegramLogo" icon={tgIcon} />
                    <IconButton alt="whatsappLogo" icon={whatsappIcon} />
                    <IconButton alt="viberLogo" icon={viberIcon} />
                    <IconButton alt="instagramLogo" icon={instagramIcon} />
                    <IconButton alt="vkLogo" icon={vkIcon} />
                </div>
            </div>
        );
    };

    return (
        <FooterContainer className={styles.footer} type={EContainerType.FOOTER}>
            {renderFooterLogo()}
            {renderMenu()}
            {renderContactsGroup()}
        </FooterContainer>
    );
};
