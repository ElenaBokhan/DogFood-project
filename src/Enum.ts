export enum ETestId {
    ADD_PRODUCT_NAME_INPUT = 'addProductNameInput',
    ADD_PRODUCT_PRICE_INPUT = 'addProductPriceInput',
    ADD_PRODUCT_DESCRIPTION_INPUT = 'addProductDescriptionInput',
    ADD_PRODUCT_DISCOUNT_INPUT = 'addProductDiscountInput',
    ADD_PRODUCT_STOCK_INPUT = 'addProductStockInput',
    ADD_PRODUCT_WIGHT_INPUT = 'addProductWightInput',
    ADD_PRODUCT_PICTURES_INPUT = 'addProductPicturesInput',
    ADD_PRODUCT_SUBMIT_BUTTON = 'addProductSubmitButton',

    ADD_REVIEW_SUBMIT_BUTTON = 'addReviewSubmitButton',
    ADD_REVIEW_TEXT_INPUT = 'addReviewTextInput',

    ADD_TO_FAVOURITES = 'addToFavouritesButton',
    REMOVE_FROM_FAVOURITES = 'removeFromFavourites',

    BREADCRUMBS_BUTTON = 'breadcrumbsButton',

    FOOTER_MAIN_LOGO = 'footerMainLogo',
    FOOTER_MENU = 'footerMenu',
    FOOTER_CONTACTS = 'footerContacts',

    HEADER_MAIN_LOGO = 'headerMainLogo',
    HEADER_ADD_NEW_ICON = 'headerAddNewProduct',
    HEADER_FAVOURITES_ICON = 'headerFavouritesIcon',
    HEADER_CART_ICON = 'headerCartIcon',
    HEADER_PROFILE_ICON = 'headerProfileIcon',
    HEADER_LOG_OUT_ICON = 'headerLogOutIcon',
    HEADER_SEARCH_INPUT = 'headerSearchInput',

    SIGN_FORM_EMAIL_INPUT = 'signFormEmailInput',
    SIGN_FORM_PASSWORD_INPUT = 'signFormPasswordlInput',
    SIGN_FORM_SUBMIT_BUTTON = 'signFormSubmitButton',

    TITLE_PAGE = 'titlePage',
    NOT_FOUND = 'notFound',

    PRODUCT_LIKE_BUTTON = 'likeButton',
    PRODUCT_TRASH_BUTTON = 'trashButton',
    PRODUCT_NAME = 'productName',
    PRODUCT_IMAGE = 'productImage',
}
