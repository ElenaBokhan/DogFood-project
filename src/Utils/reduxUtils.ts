import type {AnyAction, AsyncThunk} from '@reduxjs/toolkit';

type GenericAsyncThunk = AsyncThunk<unknown, unknown, any>;
type PendingAction = ReturnType<GenericAsyncThunk['pending']>;
type RejectedAction = ReturnType<GenericAsyncThunk['rejected']>;
type FulfilledAction = ReturnType<GenericAsyncThunk['fulfilled']>;

const hasPrefix = (action: AnyAction, prefix: string): boolean => action.type.startsWith(prefix);
const isPending = (action: PendingAction): boolean => action.type.endsWith('/pending');
const isFulfilled = (action: FulfilledAction): boolean => action.type.endsWith('/fulfilled');
const isRejected = (action: RejectedAction): boolean => action.type.endsWith('/rejected');

export const isActionPending = (prefix: string) => (action: PendingAction) => {
    return hasPrefix(action, prefix) && isPending(action);
};

export const isActionRejected = (prefix: string) => (action: RejectedAction) => {
    return hasPrefix(action, prefix) && isRejected(action);
};

export const isActionFulfilled = (prefix: string) => (action: FulfilledAction) => {
    return hasPrefix(action, prefix) && isFulfilled(action);
};
