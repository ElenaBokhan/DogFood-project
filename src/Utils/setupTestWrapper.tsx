import {router} from 'App';
import {render} from '@testing-library/react';
import {Provider} from 'react-redux';
import {createMemoryRouter, RouterProvider} from 'react-router-dom';
import {ToastContainer} from 'react-toastify';
import store from 'Store/configureStore';

interface WrapperForTestsParams {
    initialRouterEntries: string[];
}

export const setupTestsWrapper = ({initialRouterEntries}: WrapperForTestsParams) => {
    const memoryRouter = createMemoryRouter(router, {
        initialEntries: initialRouterEntries,
    });
    render(
        <Provider store={store}>
            <RouterProvider router={memoryRouter} />
            <ToastContainer autoClose={2000} position="top-right" theme="light" closeOnClick />
        </Provider>
    );

    return memoryRouter;
};
