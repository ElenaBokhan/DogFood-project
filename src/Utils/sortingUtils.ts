export const calculateOldPrice = (price: number, discount: number) => {
    return Math.floor(price - (price * discount) / 100);
};
