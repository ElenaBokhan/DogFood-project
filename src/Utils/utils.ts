import {PER_PAGE} from 'Const';
import {ESortFilter} from 'Pages/Catalog/Catalog';

export const calculateOldPrice = (price: number, discount: number) => {
    return Math.floor(price - (price * discount) / 100);
};

export const isFavourite = (likes: string[], id: string) => {
    return likes.includes(id);
};

export const getDefaultFilter = (): IClientFilter => {
    return {search: '', page: 1, perPage: PER_PAGE};
};

export const dateFormatter = new Intl.DateTimeFormat('ru-RU', {
    year: 'numeric',
    month: 'short',
    day: '2-digit',
    // hour: '2-digit',
    // minute: '2-digit',
});

export const objectHasProperty = <P extends PropertyKey>(obj: unknown, prop: P): obj is object & Record<P, unknown> => {
    return typeof obj === 'object' && !!obj && Object.hasOwn(obj, prop);
};

export const getResponse = (ms: number) => {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            if (Math.random() > 0.5) {
                resolve('success');
            } else {
                reject(new Error('error'));
            }
        }, ms);
    });
};

export const getSortingProducts = (productList: IProduct[], sortFilter: ESortFilter): IProduct[] => {
    const sortProducts: IProduct[] = [...productList];

    switch (sortFilter) {
        case ESortFilter.POPULAR:
            break;
        case ESortFilter.NEW:
            break;
        case ESortFilter.CHEAPER_FIRST:
            return sortProducts.sort((first, second) => first.price - second.price);
        case ESortFilter.EXPENSIVE_FIRST:
            return sortProducts.sort((first, second) => second.price - first.price);
        case ESortFilter.DISCOUNT:
            return sortProducts.sort((first, second) => second.discount - first.discount);
        case ESortFilter.RATING:
            break;
    }

    return sortProducts;
};
