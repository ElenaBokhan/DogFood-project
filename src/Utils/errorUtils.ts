import {objectHasProperty} from 'Utils/utils';

export const getMessageFromError = (error: unknown, defaultErrorMessage: string) => {
    if (
        objectHasProperty(error, 'data') &&
        objectHasProperty(error.data, 'message') &&
        typeof error.data.message === 'string'
    )
        return error.data.message;
    return defaultErrorMessage;
};
