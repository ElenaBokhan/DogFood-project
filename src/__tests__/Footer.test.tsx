import '@testing-library/jest-dom';
import {setupTestsWrapper} from 'Utils/setupTestWrapper';
import {render, screen} from '@testing-library/react';
import {ETestId} from 'Enum';
import {Footer} from 'Components/Footer/Footer';
import {Provider} from 'react-redux';
import store from 'Store/configureStore';

describe('Тесты для компонента Footer', () => {
    beforeEach(() => {
        setupTestsWrapper({initialRouterEntries: ['/']});
    });

    describe('Проверяем доступность элементов', () => {
        test('Проверяем доступность логотипа', () => {
            expect(screen.getByTestId(ETestId.FOOTER_MAIN_LOGO)).toBeInTheDocument();
        });

        test('Проверяем доступность меню', () => {
            expect(screen.getAllByTestId(ETestId.FOOTER_MENU)[0]).toBeInTheDocument();
        });

        test('Проверяем доступность блока с контактами', () => {
            expect(screen.getByTestId(ETestId.FOOTER_CONTACTS)).toBeInTheDocument();
        });
    });

    describe('Снапшот футера', () => {
        test('Snapshot', () => {
            const {asFragment} = render(
                <Provider store={store}>
                    <Footer />
                </Provider>
            );

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
