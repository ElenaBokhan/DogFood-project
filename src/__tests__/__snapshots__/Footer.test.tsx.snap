// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тесты для компонента Footer Снапшот футера Snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footerContainer"
  >
    <div
      class="container footer"
    >
      <div
        class="logoBlock"
      >
        <img
          alt="logo"
          data-testid="footerMainLogo"
          src="[object Object]"
        />
        <span
          class="black undefined subscribe2"
        >
          © «Интернет-магазин DogFood.ru»
        </span>
      </div>
      <ul
        data-testid="footerMenu"
      >
        <li>
          <span
            class="black undefined paragraph2"
          >
            Каталог
          </span>
        </li>
        <li>
          <span
            class="black undefined paragraph2"
          >
            Акции
          </span>
        </li>
        <li>
          <span
            class="black undefined paragraph2"
          >
            Новости
          </span>
        </li>
        <li>
          <span
            class="black undefined paragraph2"
          >
            Отзывы
          </span>
        </li>
      </ul>
      <ul
        data-testid="footerMenu"
      >
        <li>
          <span
            class="black undefined paragraph2"
          >
            Оплата и доставка
          </span>
        </li>
        <li>
          <span
            class="black undefined paragraph2"
          >
            Часто спрашивают
          </span>
        </li>
        <li>
          <span
            class="black undefined paragraph2"
          >
            Обратная связь
          </span>
        </li>
        <li>
          <span
            class="black undefined paragraph2"
          >
            Контакты
          </span>
        </li>
      </ul>
      <div
        class="contacts"
        data-testid="footerContacts"
      >
        <span
          class="black bold paragraph1"
        >
          Мы на связи
        </span>
        <span
          class="black undefined paragraph1"
        >
          8 (999) 00-00-00
        </span>
        <span
          class="black undefined subscribe1"
        >
          dogfood.ru@gmail.com
        </span>
        <div
          class="social"
        >
          <button
            class="button"
            type="button"
          >
            <img
              alt="telegramLogo"
              src="[object Object]"
            />
          </button>
          <button
            class="button"
            type="button"
          >
            <img
              alt="whatsappLogo"
              src="[object Object]"
            />
          </button>
          <button
            class="button"
            type="button"
          >
            <img
              alt="viberLogo"
              src="[object Object]"
            />
          </button>
          <button
            class="button"
            type="button"
          >
            <img
              alt="instagramLogo"
              src="[object Object]"
            />
          </button>
          <button
            class="button"
            type="button"
          >
            <img
              alt="vkLogo"
              src="[object Object]"
            />
          </button>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
